mirgecom/__init__.py

```diff
@@ -25,26 +25,23 @@
 import sys
 from warnings import warn
 
-try:
-    from mpi4py import MPI
+from mpi4py import MPI
 
-    # This code avoids slow startups due to file system locking when running with
-    # large numbers of ranks. See https://github.com/illinois-ceesd/planning/issues/27
-    # for details.
+# This code avoids slow startups due to file system locking when running with
+# large numbers of ranks. See https://github.com/illinois-ceesd/planning/issues/27
+# for details.
 
-    size = MPI.COMM_WORLD.Get_size()
+size = MPI.COMM_WORLD.Get_size()
 
-    if size >= 128:
-        for mod in ["pyopencl", "pytools", "loopy"]:
-            if mod in sys.modules:
-                warn("{} already loaded, not adjusting XDG_CACHE_HOME.".format(mod))
+if size >= 128:
+    for mod in ["pyopencl", "pytools", "loopy"]:
+        if mod in sys.modules:
+            warn("{} already loaded, not adjusting XDG_CACHE_HOME.".format(mod))
 
-        if "XDG_CACHE_HOME" in os.environ:
-            warn("XDG_CACHE_HOME already set, not adjusting it.")
-        else:
-            rank = MPI.COMM_WORLD.Get_rank()
-            username = getpass.getuser()
+    if "XDG_CACHE_HOME" in os.environ:
+        warn("XDG_CACHE_HOME already set, not adjusting it.")
```

Comment:
IMO, the expectation should be that a user should set this variable in their job script for large runs, and the warning above should teach them to do so. So this code path here IMO is the expected case and should not raise a warning.

Reply:
Removed the warning.